src/components/utils/file.js

```diff
@@ -1,16 +1,109 @@
 import { findKey } from '~utils/object'
-import { isOfMimeType } from '~utils/blob'
+import { isOfMimeType, canvasToBlob } from '~utils/blob'
 
-const defaultFileTypesAccepted = ['jpg', 'jpeg', 'png', 'pdf']
-const maxFileSizeAcceptedByAPI = 10000000 // The Onfido API only accepts files below 10 MB
-export const validateFileTypeAndSize = (
+const DEFAULT_ACCEPTED_FILE_TYPES = ['jpg', 'jpeg', 'png', 'pdf']
+const MAX_FILE_SIZE_ACCEPTED_BY_API = 10000000 // The Onfido API only accepts files below 10 MB
+const MAX_IMAGE_FILE_SIZE_ACCEPTED = 3000000
+const validateFileTypeAndSize = (
   file,
-  acceptedTypes = defaultFileTypesAccepted
+  acceptedTypes = DEFAULT_ACCEPTED_FILE_TYPES
 ) =>
   findKey(
     {
       INVALID_TYPE: (file) => !isOfMimeType(acceptedTypes, file),
-      INVALID_SIZE: (file) => file.size > maxFileSizeAcceptedByAPI,
+      INVALID_IMAGE_SIZE: (file) =>
+        file.type.match(/image.*/) && file.size > MAX_IMAGE_FILE_SIZE_ACCEPTED,
+      INVALID_SIZE: (file) => file.size > MAX_FILE_SIZE_ACCEPTED_BY_API,
     },
     (checkFn) => checkFn(file)
   )
+
+const resizeImageFile = (file, onImageResize) => {
+  const reader = new FileReader()
+  reader.onload = (readerEvent) => {
+    const image = new Image()
+    image.onload = () => {
+      const resizeTo = getDimensionsToResizeTo(image)
+      const resizedWidth = resizeTo.width
+      const resizedHeight = resizeTo.height
+      const tempCanvas = document.createElement('canvas')
+      tempCanvas.width = resizedWidth
+      tempCanvas.height = resizedHeight
+      tempCanvas
+        .getContext('2d')
+        .drawImage(image, 0, 0, resizedWidth, resizedHeight)
+      const imgDiff = {
```

Comment:
cool. Do you want to add the image *file_size* also here (before and after of course)

Reply:
Ah, yes. I can include the original file's size here. The size after I can only append later at line 100, after the resized image blob is returned.